src/altinn-app-frontend/src/store/index.ts

```diff
@@ -25,6 +25,11 @@ export const setupStore = (preloadedState?: PreloadedState<RootState>) => {
 
 export const store = setupStore();
 
+// Expose store when running in Cypress
```

Comment:
Why do we need to do this?

Reply:
Have a look here:  
[https://github.com/Altinn/app-frontend-react/blob/5430ad42b31b0edd327dd992b22237e6164d862d/test/cypress/e2e/support/app-frontend.js#L141-L149](https://github.com/Altinn/app-frontend-react/blob/5430ad42b31b0edd327dd992b22237e6164d862d/test/cypress/e2e/support/app-frontend.js#L141-L149)  
  
And for examples of usage: [https://github.com/Altinn/app-frontend-react/blob/5430ad42b31b0edd327dd992b22237e6164d862d/test/cypress/e2e/integration/app-frontend/attachments-in-group.js#L412](https://github.com/Altinn/app-frontend-react/blob/5430ad42b31b0edd327dd992b22237e6164d862d/test/cypress/e2e/integration/app-frontend/attachments-in-group.js#L412)  
  
Basically, testing the expected redux state directly lets me write a more robust test, as it doesn't test the effect of known/expected bugs, but tests the redux state where errors will cause the bugs. When reloading the page and reading data from the instance, `mapAttachments()` runs in order to find all attachments in `formData` to rebuild the state. By reading the state, the e2e test can assert that the state is the same as it was before a reload (thus testing that `mapAttachments()` work).  I could also have done this by exhaustively testing all reasonable thinkable effects of a broken state, but since they are (in principle) unlimited, testing the state directly is far more robust.

Reply:
Okay, thanks for the explanation. Could you try to sum up the "why" in the comment (not just the "what"), that would be valuable for later I think 🙂

Reply:
Yup, I agree! For this specific case I guess I didn't want to tie it to "why" in the context of attachments in repeating groups, but I can update with a better comment (although generic).